### PR TITLE
Use new Qt5 connect syntax

### PR DESCRIPTION
**Pull request by promag**
Pros&cons in https://wiki.qt.io/New_Signal_Slot_Syntax.

Note that connecting to/from overloaded slot/signal is ugly before qt 5.7 (see https://stackoverflow.com/a/16795664).

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Concept ACK. The new syntax has compile time checks, removing the need to debug impossible to find syntax errors during run time.

**Comment by laanwj** [comment]
Concept ACK (because of @MarcoFalke's reasoning)

**Comment by promag** [comment]
Thanks, I'll complete it then.

**Comment by promag** [comment]
@MarcoFalke @laanwj see new lint script, it can be improved later with other checks related to Qt.

**Comment by jonasschnelli** [comment]
Concept ACK

**Comment by fanquake** [comment]
Concept ACK

**Comment by laanwj** [comment]
> see new lint script, it can be improved later with other checks related to Qt

LGTM

**Comment by laanwj** [comment]
> - found a connection to a private slot that doesn't work with the new syntax.
> - connection to/from overloaded slot/signal is ugly before qt 5.7 (see https://stackoverflow.com/a/16795664)

does this mean it's better to postpone this to a (far) future in which we can drop pre-5.7 support?

**Comment by promag** [comment]
> does this mean it's better to postpone this to a (far) future in which we can drop pre-5.7 support?

Not really, until now just a couple of these where I have to cast the slot.

**Comment by laanwj** [comment]
oh if it can be worked around and is just a couple of cases, I agree it's not a big issue, just need to be careful to test all behavior afterwards.

**Comment by promag** [comment]
@fanquake ready to review.

**Comment by promag** [comment]
> oh if it can be worked around and is just a couple of cases

@laanwj at the moment I count 25 cases.

File: src/qt/overviewpage.cpp [hunk 1]
```diff
@@ -133,12 +133,13 @@ OverviewPage::OverviewPage(const PlatformStyle *platformStyle, QWidget *parent)
...
 
-    connect(ui->listTransactions, SIGNAL(clicked(QModelIndex)), this, SLOT(handleTransactionClicked(QModelIndex)));
+    connect(ui->listTransactions, &QListView::clicked, this, &OverviewPage::handleTransactionClicked);
 
     // start with displaying the "out of sync" warnings
     showOutOfSyncWarning(true);
-    connect(ui->labelWalletStatus, SIGNAL(clicked()), this, SLOT(handleOutOfSyncWarningClicks()));
-    connect(ui->labelTransactionsStatus, SIGNAL(clicked()), this, SLOT(handleOutOfSyncWarningClicks()));
+    connect(ui->labelWalletStatus, &QPushButton::clicked, this, &OverviewPage::handleOutOfSyncWarningClicks);
+    connect(ui->labelWalletStatus, &QPushButton::clicked, this, &OverviewPage::handleOutOfSyncWarningClicks);
```
Comment (Sjors) [inline src/qt/overviewpage.cpp]: Duplicate line

File: src/qt/transactionview.cpp [hunk 2]
```diff
@@ -175,29 +175,31 @@ TransactionView::TransactionView(const PlatformStyle *platformStyle, QWidget *pa
...
+    connect(copyAddressAction, &QAction::triggered, this, &TransactionView::copyAddress);
+    connect(copyLabelAction, &QAction::triggered, this, &TransactionView::copyLabel);
+    connect(copyAmountAction, &QAction::triggered, this, &TransactionView::copyAmount);
+    connect(copyTxIDAction, &QAction::triggered, this, &TransactionView::copyTxID);
+    connect(copyTxHexAction, &QAction::triggered, this, &TransactionView::copyTxHex);
+    connect(copyTxPlainText, &QAction::triggered, this, &TransactionView::copyTxPlainText);
+    connect(editLabelAction, &QAction::triggered, this, &TransactionView::editLabel);
+    connect(showDetailsAction, &QAction::triggered, this, &TransactionView::showDetails);
+    // Double-clicking on a transaction on the transaction history page shows details
+    connect(this, &TransactionView::doubleClicked, this, &TransactionView::showDetails);
```
Comment (Sjors) [inline src/qt/transactionview.cpp]: Note for other reviewers: this was moved from `src/qt/walletview.cpp`

**Review by Sjors — CHANGES_REQUESTED** [review]
Concept ACK. I'm a big fan of compile time checks. The new syntax also helps readability because you no longer have to guess the class name (e.g. `QTableView`) from the the variable name (e.g. `ui->tableView`).

Found a few problems with  ba34538 on macOS 10.13.6 though (Qt 5.11.1 and 5.9.6):
* application hides when I navigate to a new tab (e.g. from Overview to Transactions); maybe some event needs to be not propagated?
* all console window commands cause a segfault, so much for compile time checks :-(

Other than that I tested most `connect` instances and they all seem to work.

I ran into a bunch of existing issues, none of which are end of the world, but they're a distraction when reviewing if you don't know about them.

File: src/qt/overviewpage.cpp (hunk 1 above)
Comment (promag) [reply src/qt/overviewpage.cpp]: Removed.

**Comment by promag** [comment]
> all console window commands cause a segfault, so much for compile time checks :-(

@Sjors fixed in de93a0682.

**Comment by promag** [comment]
> all console window commands cause a segfault, so much for compile time checks :-(

@Sjors fixed in 5261ab0.

> application hides when I navigate to a new tab (e.g. from Overview to Transactions); maybe some event needs to be not propagated?

Fixed in ac8661e.

**Comment by Sjors** [comment]
Can confirm these issues are fixed in ac8661e.

**Comment by promag** [comment]
Squashed and rebased. Thanks @Sjors!

**Pull request merged**